frontend/public/components/nav/items.tsx

```diff
@@ -203,13 +234,11 @@ export const createLink = (item: PluginNavItem, rootNavLink = false): React.Reac
       Component = ResourceClusterLink;
     }
     if (Component) {
-      const { id } = item.properties;
-      const key = item.properties.componentProps.name;
-      const props = item.properties.componentProps;
+      const { id, name, ...props } = item.properties;
       if (rootNavLink) {
-        return <RootNavLink id={id} {...props} key={key} component={Component} />;
+        return <RootNavLink name={name} id={id} key={name} component={Component} {...props} />;
```

Comment:
as a `key` we should use `item.properties.uid`.

we actually pass LoadedExtension<GenericNavItem> to `createLink` so `uid` prop is accessible